subprojects/plugins/src/test/groovy/org/gradle/api/tasks/GroovyRuntimeTest.groovy

```diff
@@ -94,19 +94,23 @@ class GroovyRuntimeTest extends AbstractProjectBuilderSpec {
         with(classpath.sourceCollections[0]) {
             it instanceof Configuration
             state == Configuration.State.UNRESOLVED
-            dependencies.size() == expectedJars.size()
-            expectedJars.each { expectedJar ->
-                assert dependencies.any { it.group == "org.codehaus.groovy" && it.name == expectedJar && it.version == groovyVersion } // not sure how to check classifier
+            dependencies.size() == expectedDependencies.size()
+            expectedDependencies.each { expectedJar ->
+                assert dependencies.any {
+                    it.group == expectedGroup && it.name == expectedJar && it.version == groovyVersion
+                } // not sure how to check classifier
             }
         }
 
         where:
-        groovyVersion | classifier | expectedJars
-        "2.1.2"       | ""         | ["groovy", "groovy-ant"]
-        "2.1.2"       | "-indy"    | ["groovy", "groovy-ant"]
-        "2.5.2"       | ""         | ["groovy", "groovy-ant", "groovy-templates"]
-        "2.5.2"       | "-indy"    | ["groovy", "groovy-ant", "groovy-templates"]
-        "3.0.8"       | ""         | ["groovy", "groovy-ant", "groovy-templates", "groovy-json", "groovy-xml", "groovy-groovydoc", "groovy-astbuilder", "groovy-console", "groovy-datetime", "groovy-dateutil", "groovy-nio", "groovy-sql", "groovy-test"]
+        groovyVersion        | classifier | expectedGroup         | expectedDependencies
+        "2.1.2"              | ""         | "org.codehaus.groovy" | ["groovy", "groovy-ant"]
+        "2.1.2"              | "-indy"    | "org.codehaus.groovy" | ["groovy", "groovy-ant"]
+        "2.5.2"              | ""         | "org.codehaus.groovy" | ["groovy", "groovy-ant", "groovy-templates"]
+        "2.5.2"              | "-indy"    | "org.codehaus.groovy" | ["groovy", "groovy-ant", "groovy-templates"]
+        GroovySystem.version | ""         | "org.codehaus.groovy" | ["groovy", "groovy-ant", "groovy-templates", "groovy-json", "groovy-xml", "groovy-groovydoc", "groovy-astbuilder", "groovy-console", "groovy-datetime", "groovy-dateutil", "groovy-nio", "groovy-sql", "groovy-test"]
```

Comment:
Good idea, added.

Reply:
Hmm..., I don't understand this test now. Why does Gradle not use the system Groovy when using `GroovySystem.version`? The number of JAR files seems to be correct.

What is was expecting is to have two tests here:
- Same Groovy version, different set of JAR names -> use from dependencies
- Different Groovy version, same set of JAR names -> use from dependencies

Though this test is only testing that if the classpath only contains the `groovy` JAR, then the it will use something from a repository. So the test for having a different Groovy version, but the same set of JAR names on the classpath is missing, right?